### PR TITLE
Unify map

### PR DESCRIPTION
**Pull request by mandrews**
- Allow optional map for `load_fact`
- Ensure identity map is functioning
- Remove more dead event code

### CODE REVIEW

File: lib/masamune/actions/transform.rb [hunk 1]
```diff
@@ -77,13 +66,35 @@ def relabel_dimension(target)
...
+
+    def apply_map(map, source_files, source, target)
+      Tempfile.open('masamune') do |output|
+        FileUtils.chmod(FILE_MODE, output.path)
+        result = map.apply(source_files, output)
+        result.debug = map.debug
+        logger.debug(File.read(output)) if (source.debug || result.debug)
+        yield output, result
+      end
+    end
```
Comment (phene) [inline lib/masamune/actions/transform.rb]: Do you need to ensure the tempfile is unlinked when done?

Comment (mandrews) [inline lib/masamune/actions/transform.rb]: good call - for some reason I always think `Tempfile`s are removed when they go out of scope

Comment (mandrews) [inline lib/masamune/actions/transform.rb]:
```
$ cat tmp_test.rb
require 'tempfile'

def tmp_test(unlink: true)
  file = nil
  Tempfile.open('test') do |tmp|
    begin
      file = tmp.path
      tmp.puts 'test'
      tmp.flush
      puts File.read(tmp.path)
    ensure
      tmp.unlink if unlink
    end
  end
  file
end

puts File.read(tmp_test(unlink: true)) rescue p $!
puts File.read(tmp_test(unlink: false)) rescue p $!

$ ruby temp_test.rb
test
#<Errno::ENOENT: No such file or directory @ rb_sysopen - /var/folders/bm/bt0bd1rn43lcgnnpzyrc99tr0000gp/T/test20150731-63338-cpxgze>
test
test
```

**Comment by phene** [comment]
+1

**Pull request merged**